### ISSUE TITLE
High CPU load in background when an open tab plays animations

### ISSUE THREAD

**Issue by chrisnew** [saenzramiro/rambox#785]
Hey guys, I was wondering what sometimes keeps my CPU fans spinning and I figured out that Rambox is eating CPU whenever a tab with an animation playing is opened. I can live with such behavior, however it’s not optimal, if it’s eating also that CPU even when Rambox is no focus anymore. So maybe you guys know a way to stop the animations whenever Rambox is no longer in focus. :smile: 

### Steps to reproduce

1. Integrate for example Skype
2. Open a chat with an animated emoticon
3. Do something else while Rambox is not in focus, but keep that Skype tab opened

### Expected behavior

As Rambox loses focus, stops any animation or does something to keep the CPU load down.

### Actual behavior

Rambox process keeps eating about 25% CPU load.

### ENVIRONMENT

> Rambox: v0.5.3
> OS: Arch Linux
> Arch: x86_64

`XDG_SESSION_TYPE=xwayland` (although Rambox is rendered through xwayland)

I use GNOME 3.22.2-1.

Intel integrated GPU, MacBook Air 5,1.

**Comment by NotAFile** [comment]
I agree, AFAIK webkit has an option for this, but rambox does not enable it.

**Comment by chrisnew** [comment]
Also happening on regular X11 sessions.

**Comment by yschroeder** [comment]
:+1: My rambox constantly eats one full (virtual) CPU core of my 2 two physical cores with SMT. This is especially annoying on a laptop computer and drains the battery a lot!

**Comment by ffissore** [comment]
I've just experienced the same behaviour in Rambox 0.7.0. I have a gif in a tweetdeck tab. The tab is not opened at rambox start, so the CPU stays low. As soon as I switch to the tweetdeck tab, CPU jumps at 20% and doesn't go down any more, even when switching to another tab.
The process involved has this command line 
```
/opt/Rambox/rambox --type=renderer --field-trial-handle=14876871545345889055,256617573391749382,131072 --disable-features=SpareRendererForSitePerProcess --lang=it --app-path=/opt/Rambox/resources/app.asar --enable-plugins --no-sandbox --no-zygote --preload=/opt/Rambox/resources/app.asar/resources/js/rambox-service-api.js --background-color=#fff --guest-instance-id=10 --enable-blink-features --disable-blink-features --num-raster-threads=4 --enable-main-frame-before-activation --service-request-channel-token=16398263671307416608 --renderer-client-id=26 --no-v8-untrusted-code-mitigations --shared-files=v8_context_snapshot_data:100,v8_natives_data:101
```

and it's paired with this one

```
/opt/Rambox/rambox --type=gpu-process --field-trial-handle=14876871545345889055,256617573391749382,131072 --disable-features=SpareRendererForSitePerProcess --gpu-preferences=KAAAAAAAAAAgAAAgAAAAAAAAYAAAAAAAEAAAAAAAAAAAAAAAAAAAAAgAAAAAAAAA --service-request-channel-token=15453922080006969632
```

**Comment by fvulich** [comment]
Is this still reproducible?

**Comment by ffissore** [comment]
Yes. You need a service that shows an animated GIF
As soon as the GIF is displayed on that service, rambox starts eating CPU, even if you switch to a different service: it looks like it's still rendering it, even if not displayed. Proof is that, if you switch back to that service and do something to "hide" that GIF (like scrolling down a twitter stream), CPU usage drops
